File: 05_Elixir/workers/lib/incoming/execution_model.ex
defmodule Incoming.ExecutionModel do
  @type t ::
          :one_off
          | :scheduled
          | :event_driven
          | :batch
          | :stream
          | :workflow
          | :actor

  def one_off, do: :one_off
  def scheduled, do: :scheduled
  def event_driven, do: :event_driven
  def batch, do: :batch
  def stream, do: :stream
  def workflow, do: :workflow
  def actor, do: :actor

  def all,
    do: [
      :one_off,
      :scheduled,
      :event_driven,
      :batch,
      :stream,
      :workflow,
      :actor
    ]
end

File: 06_Kotlin/worker/src/main/kotlin/incoming/ExecutionModel.kt
package org.andreastme.langpermonth.incoming

enum class ExecutionModel {
    OneOff,         // Single fire-and-forget unit
    Scheduled,      // Time-triggered (cron/delayed)
    EventDriven,    // Triggered by domain/infrastructure event
    Batch,          // Large dataset processing
    Stream,         // Long-lived continuous processing
    Workflow,       // DAG / multistep orchestration
    Actor;          // Key-partitioned stateful unit

    companion object {
        fun from(value: Int): ExecutionModel {
            return ExecutionModel.entries.first { it.ordinal == value }
        }
    }
}